core/app/models/concerns/spree/active_storage_adapter/attachment.rb

```diff
@@ -53,8 +53,9 @@ def destroy
 
       def metadata
         analyze unless analyzed?
-
         @attachment.metadata
+      rescue ActiveStorage::FileNotFoundError
+        { identified: nil, width: nil, height: nil, analyzed: false }
```

Comment:
@elia @filippoliverani any thoughts on this? I can't think of a better way to fix this problem but maybe we already faced something similar or have other ideas?

Reply:
@elia @filippoliverani pinging you again to move it forward :slightly_smiling_face:

Reply:
The solution makes sense to me, but I'd prefer to let the error bubble up and handle it where the attachment is used. That's a breaking error, and different controllers or views might want to handle it in different ways, returning a null object might be misleading 🤔

Reply:
I'm not sure about this solution, but please correct me if I'm wrong. If there are 20 images on the page and only one of them is unavailable because of this problem, bubbling up the exception to the controller/view layer we are not able to display the other 19 images. We can only show an error to the user.

Reply:
Don't you think that could be the desired behavior sometimes? Depending on the model, you might want to catch that issue instead of silencing it. For instance, as an administrator, you want to notice when you upload an image for a product, but it's corrupted.

But I made an error when I said, "we could manage it at the controller layer with a configurable error handler". We should just make returning a null object an optional behavior. If someone asks for it, we can do more work and make it configurable on a model basis.

Reply:
My opinion here is that, if this failure will prevent people to buy things from the store, I think that showing a broken image is a better solution. So maybe storefront and APIs should never break because of this. 

On the other side, the admin should notice that there's something wrong, but if we just show the error, what can they do to fix it? I'm afraid that they need to ask some devs to manually destroy the record and try to reupload the image, and probably it's better to keep the admin working until that happens.

Reply:
> My opinion here is that, if this failure will prevent people to buy things from the store, I think that showing a broken image is a better solution. So maybe storefront and APIs should never break because of this.

What I meant is: A user wants to buy a shirt, but, as the image is not rendered, they don't buy it. As no error is logged, the admin will never know about the problem unless they revisit the product page for some reason.

> On the other side, the admin should notice that there's something wrong, but if we just show the error, what can they do to fix it?

That's an excellent point. Admin shouldn't break. We need different behavior depending on the component.

A good middle-way solution could be returning a nil image but showing a warning so that it's logged and can be detected. What do you think?

Reply:
Yes, or delete the corrupted image and continue with the execution. Could be dangerous, I don't know the exact implications of this proposal but if I got it correctly, admin will need to reupload the image again in this case because it's not saved in the storage provider. Why keep an instance in our system for that image the actual image is nowhere?

Reply:
Not sure, but maybe the stored metadata could help debug?

Reply:
Probably, I don't even know if this is reproducible, to be honest.

Reply:
@cpfergus1, when you have a moment, could you please add a warning when the image is not found? 🙏

Reply:
Hey, @cpfergus1, pinging you again in case you have time to put the little extra effort needed to complete this 💪

Reply:
Sorry, I missed the request, I will pick it back up!

Reply:
I have added a point to capture this error in the logs, did we still want this to bubble up to the controller?

Reply:
Thanks, @cpfergus1! No, I think it's better not to bubble up to the controller at this point.